app/modules/notifications/models.py

```diff
@@ -30,24 +37,69 @@ class Notification(db.Model, HoustonModel):
     guid = db.Column(
         db.GUID, default=uuid.uuid4, primary_key=True
     )  # pylint: disable=invalid-name
-    timestamp = db.Column(db.DateTime, server_default=func.now(), nullable=False)
+
     status = db.Column(
         db.String(length=255), default=NotificationStatus.unread, nullable=False
     )
-    message_template = db.Column(
-        db.String, default=NotificationMessageTemplate.raw, nullable=False
-    )
+    message_type = db.Column(db.String, default=NotificationType.raw, nullable=False)
     message_values = db.Column(db.JSON, nullable=True)
     recipient_guid = db.Column(
         db.GUID, db.ForeignKey('user.guid'), index=True, nullable=True
     )
-    recipient = db.relationship('User')
+    recipient = db.relationship('User', back_populates='notifications')
 
     def __repr__(self):
         return (
             '<{class_name}('
             'guid={self.guid}, '
-            'template={self.message_template}, '
+            'template={self.message_type}, '
             "recipient='{self.recipient}'"
             ')>'.format(class_name=self.__class__.__name__, self=self)
         )
+
+    # returns dictionary of channel:bool
+    def channels_to_send(self):
+        # For now is_digest is ignored, that's implemented post MVP
+        # pylint: disable=invalid-name
+        from app.modules.site_settings.models import SiteSetting
+
+        user_prefs = [
+            pref
+            for pref in self.recipient.notification_preferences
+            if pref.message_type == self.message_type
```

Comment:
suspect bugs here in terms of comparing enum w string of enum but that's debug rather than design.